#
# old_revision [20e2add42a314b4914e798757eb2a69ac3b491d0]
#
# patch "libpurple/protocols/jabber/jabber.c"
#  from [37a356263832da85c1ce5dac41c9c1204cb8f407]
#    to [27f982381ff26d8e01a10a7d068536a9f5c95ce9]
# 
# patch "libpurple/protocols/jabber/jabber.h"
#  from [9f3858e49283499a451c4f78dff6a1d782c20467]
#    to [b1a7f0f55653701e2ef512f726ab89371b8aa9b3]
#
============================================================
--- libpurple/protocols/jabber/jabber.c	37a356263832da85c1ce5dac41c9c1204cb8f407
+++ libpurple/protocols/jabber/jabber.c	27f982381ff26d8e01a10a7d068536a9f5c95ce9
@@ -536,7 +536,7 @@ static void tls_init(JabberStream *js)
 	purple_input_remove(js->gc->inpa);
 	js->gc->inpa = 0;
 	js->gsc = purple_ssl_connect_with_host_fd(js->gc->account, js->fd,
-			jabber_login_callback_ssl, jabber_ssl_connect_failure, js->serverFQDN, js->gc);
+			jabber_login_callback_ssl, jabber_ssl_connect_failure, js->certificate_CN, js->gc);
 }
 
 static void jabber_login_connect(JabberStream *js, const char *fqdn, const char *host, int port)
@@ -591,7 +591,8 @@ jabber_login(PurpleAccount *account)
 	js->next_id = g_random_int();
 	js->write_buffer = purple_circ_buffer_new(512);
 	js->old_length = -1;
-
+	js->certificate_CN = g_strdup(connect_server[0] ? connect_server : js->user->domain);
+	
 	if(!js->user) {
 		purple_connection_error_reason (gc,
 			PURPLE_CONNECTION_ERROR_INVALID_SETTINGS,
@@ -628,7 +629,7 @@ jabber_login(PurpleAccount *account)
 	if(purple_account_get_bool(js->gc->account, "old_ssl", FALSE)) {
 		if(purple_ssl_is_supported()) {
 			js->gsc = purple_ssl_connect(js->gc->account,
-					connect_server[0] ? connect_server : js->user->domain,
+					js->certificate_CN,
 					purple_account_get_int(account, "port", 5223), jabber_login_callback_ssl,
 					jabber_ssl_connect_failure, js->gc);
 		} else {
@@ -1103,6 +1104,7 @@ void jabber_register_account(PurpleAccou
 		my_jb->subscription |= JABBER_SUB_BOTH;
 
 	server = connect_server[0] ? connect_server : js->user->domain;
+	js->certificate_CN = g_strdup(server);
 
 	jabber_stream_set_state(js, JABBER_STREAM_CONNECTING);
 
@@ -1279,6 +1281,7 @@ void jabber_close(PurpleConnection *gc)
 		js->commands = g_list_delete_link(js->commands, js->commands);
 	}
 	g_free(js->server_name);
+	g_free(js->certificate_CN);
 	g_free(js->gmail_last_time);
 	g_free(js->gmail_last_tid);
 	g_free(js->old_msg);
============================================================
--- libpurple/protocols/jabber/jabber.h	9f3858e49283499a451c4f78dff6a1d782c20467
+++ libpurple/protocols/jabber/jabber.h	b1a7f0f55653701e2ef512f726ab89371b8aa9b3
@@ -191,6 +191,8 @@ struct _JabberStream
 	char *old_uri;
 	int old_length;
 	char *old_track;
+	
+	char *certificate_CN;
 };
 
 typedef gboolean (JabberFeatureEnabled)(JabberStream *js, const gchar *shortname, const gchar *namespace);
